### ISSUE TITLE
Unable to set values in Excel range when values start with a minus.

### ISSUE THREAD

**Issue by pvsleeper** [OfficeDev/office-js#3256]
# Provide required information needed to triage your issue
When setting Excel range values and the first item in the array starts with a minus and the string is large, the sync operation fails with a "There was an internal error while processing the request" error

## Your Environment
* Platform [PC desktop, Mac, iOS, Office on the web]: PC Desktop
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: 2208 (Build  15601.20578)
* Operating System: Windows 10 Enterprise
* Browser (if using Office on the web): ______
 
## Expected behavior
The values in the range should get updated.

## Current behavior
Currently when setting the values and the first item in the array starts with a **minus** and the string is fairly large (appears to start happening around the 8200 char area), the setting action errors and throws an exception. 

When removing the minus, the operation completes successfully. Also reducing the string length also makes it succeed (the number of characters where the issue appears varies based on the content of the string from what I've seen so I am not convinced always using a smaller string will address this issue) 

## Steps to reproduce
Run the following in ScriptLab:
```
async function run() {
  await Excel.run(async (context) => {
    const sheet = context.workbook.worksheets.getActiveWorksheet();
    const range = sheet.getCell(0,0).getColumnsAfter(2);
    const values = [
      "-" + ("X").repeat(8200), 
      "This can be any string or an empty string, but we need at least 2 items in the array, then we get the crash"
    ];
    range.values = [values]; 
    await context.sync();
  });
}
```
In the above snippet:
- If you update the first member of the values array NOT to start with a "-", the values gets updated successfully. 
- If you update the first member of the values array to be a string thats around 8100 chars, the values gets updated successfully. (The length appears to vary based on the contents of the string) 
- If you remove the 2nd item in the array and adjust the range to be only 1 column, then the value gets updated successfully.
- The ~8200 chars is way below the 32k limit for cells.
- Changing it to set the range.formulas instead of range.values has no impact.

## Context
I am trying to serialize data to a hidden sheet that might span over multiple columns. Currently have it to split the data in 32k chunks to stick to the cell limit, but then discovered this. 
This issue is preventing me from reliably saving the data to the hidden sheet.

## Useful logs
![image](https://user-images.githubusercontent.com/15993009/227841401-575af503-5d93-4877-8f9a-5b4c517f77b3.png)

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by honxmsft** [comment]
@pvsleeper We've repro this issue and opened a bug to track this issue. Will try to fix the bug accordingly.

I wonder if splitting the data into smaller size will solve the issue? (Like 16k)

**Comment by pvsleeper** [comment]
@honxmsft - thank you for logging the bug! Even at just over 8k the issue still happens. My interim workaround will be to add a space in front of every set of values. That seems to get around what ever it is thats unhappy with the "-" and then when I re-assemble the JSON, I strip out the first char.

**Comment by ruijiaMS** [comment]
Hi @pvsleeper,
Thanks for report this issue! When you are setting values into a cell with the start of some specific characters like +/-/= and so on, Excel will first check the given character can start a formula. That means if you add the "-" as the first character, the cell actually should be a candidate formula. Meanwhile another check is the length of the value. As the length of your input is 8200 which is higher than a threshold. So it failed under these two check. We consider this error is a benign exception which can help user to avoid unexpected input and optimize their inputs.
So basically I provide two options for you:
1, do not use these specific characters as the first character.
2, shorten the length less than 8000.

Hope this helped you!

**Comment by pvsleeper** [comment]
Hi @ruijiaMS - a more descriptive error would really help. Took me forever to isolate it. 

And none of the documentation states any of these restrictions. Will you be updating the documentation to state that the cell limits are 32k, unless it happens to start with a "-" then it becomes some random just over 8k max length?

Since there is no other way to store metadata (all the MS documentation states that using hidden sheets is the way to do it, especially if you're trying to support co-authoring), it its really important to make consumers of the API understand what the limits are.

**Comment by ruijiaMS** [comment]
Hi @pvsleeper 
Sorry for the unclear description in our doc, and I totally agree with you that we should provider more helpful information of API, especially for these cases. We have an internal bug 7734442 to tack this doc update. Thanks for your report and will let you know when we complete the update.

**Comment by pvsleeper** [comment]
Thank you @ruijiaMS

**Comment by ruijiaMS** [comment]
@pvsleeper could you please have a try with this workaround?
```
await Excel.run(async (context) => {
    const sheet = context.workbook.worksheets.getActiveWorksheet();
    let range = sheet.getCell(0, 0).getColumnsAfter(1);
    let values = ["-" + ("X").repeat(8200),];
    range.values = [values];

    range = range.getColumnsAfter(1);
    values = ["This can be any string or an empty string, but we need at least 2 items in the array, then we get the crash",]
    range.values = [values];
    await context.sync();
  });
```

**Comment by pvsleeper** [comment]
@ruijiaMS - yes, that will definitely work, since the issue only seems to happen if I am setting an array with at least 2 values.
But, according to the documentation, this is not the preferred way, as this will add overhead with extra api commands in the queue and the recommended way is to do it with a single call

**Comment by ruijiaMS** [comment]
@pvsleeper Got it! The work item 7734442 is on track.

**Comment by XuanZhouMSFT** [comment]
Thanks for reporting this issue. After careful consideration, our team has decided to not fix this issue in the short term. We recommend you to upvote or submit a new Tech Community New Ideas item, this will give us visibility of the commonality of it and will help with our prioritization. If we end up working on this issue, we will re-activate as needed.
